mmedit/models/backbones/sr_backbones/dfd_net.py

```diff
@@ -0,0 +1,671 @@
+            nn.LeakyReLU(0.2, True),
+            spectral_norm(nn.Conv2d(out_channel, out_channel, 3, 1, 1)),
+            nn.Sigmoid())
+
+    def forward(self, input, style):
+        """Forward Function.
+
+        Args:
+            input (Tensor): Input tensor.
+            style (Tensor): Style tensor.
+
+        Returns:
+            Tensor: Forward results.
+        """
+
+        out = self.conv_first(input)
+        shift = self.shift_model(style)
+        scale = self.scale_model(style)
+        out = out * scale + shift
+        out_up = self.conv_up(out)
+
+        return out_up
+
+
+class VGGFeat(torch.nn.Module):
+    """Encoder based on VGG.
+    """
+
+    def __init__(self, load_path):
+        super().__init__()
+        self.model = models.vgg19(pretrained=False)
+        if load_path:
+            self.model.load_state_dict(torch.load(load_path))
+        self.build_vgg_layers()
+
+        self.register_parameter(
+            'rgb_mean',
+            nn.Parameter(torch.Tensor([0.485, 0.456, 0.406]).view(1, 3, 1, 1)))
+        self.register_parameter(
+            'rgb_std',
+            nn.Parameter(torch.Tensor([0.229, 0.224, 0.225]).view(1, 3, 1, 1)))
+
+        for param in self.model.parameters():
+            param.requires_grad = False
+
+    def build_vgg_layers(self):
+        """Build VGG Layers.
+        """
+        vgg_pretrained_features = self.model.features
+        self.features = []
+        feature_layers = [0, 8, 17, 26, 35]
+        for i in range(len(feature_layers) - 1):
+            module_layers = torch.nn.Sequential()
+            for j in range(feature_layers[i], feature_layers[i + 1]):
+                module_layers.add_module(str(j), vgg_pretrained_features[j])
+            self.features.append(module_layers)
+        self.features = torch.nn.ModuleList(self.features)
+
+    def preprocess(self, x):
+        """Modify the mean and variance.
+
+        Args:
+            x (Tensor): Input tensor.
+
+        Results:
+            Tensor: Processed tensor.
+        """
+        # print('save')
+        # torch.save(self.state_dict(), 'work_dirs/vgg_load.pth')
+        x = (x + 1) / 2
+        x = (x - self.rgb_mean) / self.rgb_std
+        if x.shape[3] < 224:
+            x = torch.nn.functional.interpolate(
+                x, size=(224, 224), mode='bilinear', align_corners=False)
+        return x
+
+    def forward(self, x):
+        """Forward Function.
+
+        Args:
+            x (Tensor): Input tensor.
+
+        Results:
+            Tensor: Forward results.
+        """
+
+        x = self.preprocess(x)
+        features = []
+        for m in self.features:
+            x = m(x)
+            features.append(x)
+        return features
+
+
+@BACKBONES.register_module()
+class DFDNet(nn.Module):
+    """DIC network structure for face super-resolution.
+
+    Paper: Blind Face Restoration via Deep Multi-scale Component Dictionaries.
+
```

Comment:
If there is a repo that this implementation is referred heavily, then add its info here

Reply:
Ref repo -> Reference implementation